gcore/gdalmultidim.cpp

```diff
@@ -1692,6 +1695,9 @@ bool GDALExtendedDataType::CopyValue(const void *pSrc,
                                  static_cast<GIntBig>(
                                      *static_cast<const std::int64_t *>(pSrc)));
                 break;
+            case GDT_Float16:
+                str = CPLSPrintf("%.9g", *static_cast<const _Float16 *>(pSrc));
```

Comment:
I would expect float16 requires less than 9 significant digits. Same below for CFloat16

Reply:
Thanks, it requires 5 digits.

I notice that `double` values are output with 18 digits. 17 would suffice.